pkg/provider/consulcatalog/consul_catalog.go

```diff
@@ -451,17 +455,11 @@ func (p *Provider) watchConnectTLS(ctx context.Context, leafWatcher *watch.Plan,
 		}
 	}()
 
-	leafCerts := <-leafChan
```

Comment:
Same as in the above comment, moved it down into the main loop so `Done()` is taken into account.